sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.net8.0.cs

```diff
@@ -432,6 +489,109 @@ protected virtual void JsonModelWriteCore(System.Text.Json.Utf8JsonWriter writer
+        Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.Create(System.BinaryData data, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+    }
+    public partial class MaintenanceWindowSchedule : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>
+    {
+        public MaintenanceWindowSchedule() { }
+        public Azure.ResourceManager.RedisEnterprise.Models.MaintenanceDayOfWeek? DayOfWeek { get { throw null; } set { } }
+        protected virtual void JsonModelWriteCore(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options) { }
+        Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>.Create(ref System.Text.Json.Utf8JsonReader reader, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        void System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>.Write(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options) { }
+        Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>.Create(System.BinaryData data, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct MaintenanceWindowType : System.IEquatable<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowType>
```

Comment:
name too generic

Reply:
Renamed to `ClusterCustomMaintenanceWindowType`